src/app/reporting/Engine.cpp

```diff
@@ -773,31 +773,32 @@ CHIP_ERROR Engine::SetDirty(AttributePathParams & aAttributePath)
 {
     BumpDirtySetGeneration();
 
-    InteractionModelEngine::GetInstance()->mReadHandlers.ForEachActiveObject([&aAttributePath](ReadHandler * handler) {
-        // We call SetDirty for both read interactions and subscribe interactions, since we may sent inconsistent attribute data
-        // between two chunks. SetDirty will be ignored automatically by read handlers which is waiting for response to last message
-        // chunk for read interactions.
-        if (handler->IsGeneratingReports() || handler->IsAwaitingReportResponse())
-        {
-            for (auto object = handler->GetAttributePathList(); object != nullptr; object = object->mpNext)
+    bool hasOverlappedPath = false;
+    InteractionModelEngine::GetInstance()->mReadHandlers.ForEachActiveObject(
+        [&aAttributePath, &hasOverlappedPath](ReadHandler * handler) {
+            // We call SetDirty for both read interactions and subscribe interactions, since we may sent inconsistent attribute data
```

Comment:
```suggestion
            // We call SetDirty for both read interactions and subscribe interactions, since we may send inconsistent attribute data
```

Reply:
done, thanks